lib/testing/integration/authservice.go

```diff
@@ -143,23 +143,25 @@ func (auth *AuthService) Run(ctx context.Context) error {
 		stdout := bufio.NewReader(stdoutPipe)
 		for {
 			line, err := stdout.ReadString('\n')
-			if line != "" {
-				auth.saveStdout(line)
-				auth.parseLine(ctx, line)
-				if !auth.IsReady() {
-					if addr := auth.AuthAddr(); !addr.IsEmpty() {
-						log.Debugf("Found addr of Auth service process: %v", addr)
-						auth.setReady(true)
-					}
-				}
-			}
 			if err == io.EOF {
 				return
 			}
 			if err := trace.Wrap(err); err != nil {
 				log.WithError(err).Error("failed to read process stdout")
 				return
 			}
+
+			auth.saveStdout(line)
+
+			if auth.IsReady() {
+				continue
```

Comment:
Is this correct? Won't this put us in a busy loop if auth is ready?

Reply:
It shouldn't, unless I'm missing something.

The first line of the loop calls `line, err := stdout.ReadString('\n')` which should block until a new line is returned (it could be an empty one, but that shouldn't matter for this use case)